### PR TITLE
Use soft reboot/shutdown by default and provide hard reboot/shutdown actions

### PR DESCRIPTION
**Pull request by vuntz**
Note that the hard reboot/shutdown methods are not exposed in the UI.

### CODE REVIEW

**Comment by jsuchome** [comment]
Nice. But yes, we should have it in UI (probably in some other request).

I do not really like the name, 'hard reboot'. Wouldn't 'force reboot' be better? Or maybe it is usually called this way and my language feeling is wrong.

**Comment by vuntz** [comment]
@jsuchome http://en.wikipedia.org/wiki/Reboot_%28computing%29 :-)

**Comment by jsuchome** [comment]
Not that it really matters, but that article defines cold/hard reboot as "when power to the system is physically turned off and back on again". Which is not what ``reboot -f`` does.

**Comment by vuntz** [comment]
Btw, the reason I haven't put that in the UI is that I don't know how to add the buttons without having it look ugly because of too many buttons.

**Comment by vuntz** [comment]
> Not that it really matters, but that article defines cold/hard reboot as "when power to the system is physically turned off and back on again". Which is not what ``reboot -f`` does.

Well, when IPMI is used, that's what it does. But you're right, I guess. I'm happy to rename to force reboot/shutdown if others agree it's better.

**Comment by jsuchome** [comment]
Yes, it probably needs a touch from UI guys :-)

**Comment by tboerger** [comment]
Maybe we can add a button group with a drodown list for all these actions. That way it stays intact even for responsive small screen widths.

**Comment by vuntz** [comment]
@tboerger big +1. No idea how to do that, though :-)

**Comment by tboerger** [comment]
```html
<div class="dropdown">
  <button class="btn btn-default dropdown-toggle" type="button" data-toggle="dropdown" aria-expanded="true">
    Actions
    <span class="caret"></span>
  </button>
  <ul class="dropdown-menu" role="menu">
    <li role="presentation"><a role="menuitem" href="#">Action</a></li>
    <li role="presentation"><a role="menuitem" href="#">Another action</a></li>
    <li role="presentation"><a role="menuitem" href="#">Something else here</a></li>
    <li role="presentation"><a role="menuitem" href="#">Separated link</a></li>
  </ul>
</div>
```

**Comment by jsuchome** [comment]
@tboerger could  you reformulate it to haml? Currently we have https://github.com/crowbar/barclamp-crowbar/blob/master/crowbar_framework/app/views/nodes/_buttons_power.html.haml

**Comment by tboerger** [comment]
```ruby
.dropdown
  %button.btn.btn-default.dropdown-toggle{ type: "button", data: { toggle: "dropdown" } }
    = t(".actions")
    %span.caret

  %ul.dropdown-menu{ role: "menu" }
    - if @node.admin?
      - if @node.bmc_set?
        %li{ role: "presentation" }
          = link_to t(".identify"), hit_node_path(@node.handle, "identify"), role: "menuitem", remote: true
    - else
      %li{ role: "presentation" }
        = link_to t(".reboot"), hit_node_path(@node.handle, "reboot"), role: "menuitem", remote: true
      %li{ role: "presentation" }
        = link_to t(".hardreboot"), hit_node_path(@node.handle, "hardreboot"), role: "menuitem", remote: true
      %li{ role: "presentation" }
        = link_to t(".shutdown"), hit_node_path(@node.handle, "shutdown"), role: "menuitem", remote: true
      %li{ role: "presentation" }
        = link_to t(".hardshutdown"), hit_node_path(@node.handle, "hardshutdown"), role: "menuitem", remote: true

      - if @node.bmc_set?
        %li{ role: "presentation" }
          = link_to t(".poweron"), hit_node_path(@node.handle, "poweron"), role: "menuitem", remote: true
        %li{ role: "presentation" }
          = link_to t(".identify"), hit_node_path(@node.handle, "identify"), role: "menuitem", remote: true
```

I have not tested it, so please check if the ```data-remote``` attributes are really set and if it looks correct. Don't miss the i18n keys :).

File: crowbar_framework/app/models/node_object.rb [hunk 1]
```diff
@@ -1169,7 +1176,7 @@ def set_state(state)
         Rails.logger.warn("chef client seems to be still running after 5 minutes of wait; going on with the reboot")
       end
 
-      bmc_cmd("power cycle")
+      ssh_cmd("/sbin/reboot")
```
Comment (dirkmueller) [inline crowbar_framework/app/models/node_object.rb]: that doesn't work on windows :-)

Comment (tboerger) [inline crowbar_framework/app/models/node_object.rb]: Correct, just call shitdown -r without a path.

**Comment by vuntz** [comment]
I renamed the hard reboot/hard shutdown bits to power cycle/power off, as it works nicely with the "power on" bit.

**Comment by aplanas** [comment]
LGTM if is expected to work only on Linux (calls to /sbin/* commands and references to /var/run/crowbar/chef-client.* files)

Maybe we can add a check to disable this command in Windows (node[:platform] == 'windows') in _buttons_power.html.haml

**Comment by jsuchome** [comment]
+1

File: crowbar_framework/app/models/node_object.rb (hunk 1 above)
Comment (tboerger) [inline crowbar_framework/app/models/node_object.rb]: What about this? If we call it only ```shutdown -r``` it should work on windows and linux platform.

Comment (vuntz) [inline crowbar_framework/app/models/node_object.rb]: There's no ssh server on the hyperv nodes, so no, it won't work.

Comment (tboerger) [inline crowbar_framework/app/models/node_object.rb]: So we should hide the buttons for windows machines.

Comment (vuntz) [inline crowbar_framework/app/models/node_object.rb]: Yes, but it's unrelated to this PR. Doing another PR for it right now :-)

Comment (vuntz) [inline crowbar_framework/app/models/node_object.rb]: Done in https://github.com/crowbar/barclamp-crowbar/pull/1202

**Pull request merged**